src/views/admin/AllOperations.vue

```diff
@@ -75,32 +77,34 @@
             }
 
             async function getOperations() {
+                const promises = [];
                 const operationManagement = new OperationManagement();
                 const handler = new GenericResponseHandler("operations");
-                const response = await operationManagement.getOperations(undefined, undefined, undefined, false);
-                const result = handler.handleResponse(response);
 
-                //Show empty list if no results given
-                operations.value = result;
-                gotOps.value = true;
-                await getWatchedOperations();
-            }
+                promises.push(
+                    operationManagement.getOperations(undefined, undefined, undefined, false).then((response) => {
+                        const result = handler.handleResponse(response);
+                        operations.value = result;
+                    })
+                );
 
-            async function getWatchedOperations() {
-                const operationManagement = new OperationManagement();
-                const handler = new GenericResponseHandler("operations");
-                const response = await operationManagement.getOperations(undefined, undefined, undefined, true);
-                const result = handler.handleResponse(response);
-                const ownId = await getOwnEnrollmentId();
-                watchedOperations.value = result.filter((op) => op.initiator != ownId);
-            }
+                let tmpOps = [] as Operation[];
+                promises.push(
+                    operationManagement.getOperations(undefined, undefined, undefined, true).then((response) => {
+                        tmpOps = handler.handleResponse(response);
+                    })
+                );
 
-            async function getOwnEnrollmentId(): Promise<string> {
                 const certificateManagement = new CertificateManagement();
-                const handler = new GenericResponseHandler("enrollment-id");
-                const response = await certificateManagement.getOwnEnrollmentId();
-                const result = handler.handleResponse(response);
-                return result.id;
+                let ownId = "";
+                promises.push(
+                    certificateManagement.getOwnEnrollmentId().then((response) => {
+                        ownId = handler.handleResponse(response).id;
+                    })
+                );
+                await Promise.all(promises);
+                gotOps.value = true;
+                watchedOperations.value = tmpOps.filter((op) => op.initiator != ownId);
```

Comment:
correct me if I am wrong, but tmpOps is only used if you are not the initiator? So you can just use

`operationManagement.getOperations(false, undefined, undefined, true)`

the first parameter is selfInitiated

Reply:
Removed in https://github.com/upb-uc4/ui-web/pull/826/commits/8b3cb54ee4dbed633cfbe1408263a925c3926695